### ISSUE TITLE
E2guardian doesn't read the last line of config files

### ISSUE THREAD

**Issue by kalioz** [e2guardian/e2guardian#592]
this line https://github.com/e2guardian/e2guardian/blob/2342c6af4f88d040d7a191943e4bf606b57fe10c/src/FOptionContainer.cpp#L159 makes it so that the config parser doesn't read the line if the file has been read entirely.

When the file doesn't end with a newline this means the last line isn't parsed.
e.g. : without newline:
```
htmltemplate = 'template.json'
neterrtemplate = 'neterr_template.json' # this won't be parsed
```
with newline : 
```
htmltemplate = 'template.json'
neterrtemplate = 'neterr_template.json' # this will be parsed
# this won't be parsed
```

replacing this line by
 ```c
if (linebuffer.length() != 0) {
```
should enable it to parse the last line.

I've never touched at c++ and don't have the time to test this (if this exists there might be a reason ? ), so if someone wants to take a look at it / explain why this exists in the docs you're welcome to.

**Comment by philipianpearce** [comment]
OK, yes there is a bug here - however the last line is ignored in all cases.

I will fix this.

But if the last line is not terminated with newline the return line from getline will be indeterminate so cannot be used.  i.e. All config files must end with a newline and if not anything after the final newline will be ignored.

This bug goes back a long time - thanks for finding it!

**Comment by philipianpearce** [comment]
Fixed in v5.3.dev and v5.4.dev.

**Comment by kalioz** [comment]
damn, that was fast ! 
thank you for fixing it :)
